homeassistant/components/heos/quality_scale.yaml

```diff
@@ -0,0 +1,82 @@
+rules:
+  # Bronze
+  action-setup: done
+  appropriate-polling:
+    status: done
+    comment: Integration is a local push integration
+  brands: done
+  common-modules: todo
+  config-flow-test-coverage: done
+  config-flow: done
+  dependency-transparency: done
+  docs-actions: done
+  docs-high-level-description: done
+  docs-installation-instructions: done
+  docs-removal-instructions: todo
+  entity-event-setup: done
+  entity-unique-id: done
+  has-entity-name: done
+  runtime-data: done
+  test-before-configure: todo
+  test-before-setup: done
+  unique-config-entry:
+    status: todo
+    comment: |
+      The HEOS integration only supports a single config entry, but needs to be migrated to use
+      the `single_config_entry` flag. HEOS devices interconnect to each other, so connecting to
+      a single node yields access to all the devices setup with HEOS on your network. The HEOS API
+      documentation does not recommend connecting to multiple nodes which would provide no bennefit.
+  # Silver
+  action-exceptions: todo
+  config-entry-unloading: done
+  docs-configuration-parameters:
+    status: done
+    comment: |
+      The integration doesn't provide any additional configuration parameters.
+  docs-installation-parameters: done
+  entity-unavailable: done
+  integration-owner: done
+  log-when-unavailable:
+    status: todo
+    comment: |
+      The integration currently spams the logs until reconnected
+  parallel-updates:
+    status: todo
+    comment: Needs to be set to 0. The underlying library handles parallel updates.
+  reauthentication-flow:
+    status: exempt
+    comment: |
+      This integration doesn't require re-authentication.
+  test-coverage: done
```

Comment:
`test_updates_from_players_changed_new_ids` -> So If I understand correctly, the identifiers of the devices can change on runtime? Why do they do that?

Reply:
> identifiers of the devices can change on runtime? Why do they do that?

Yes, they can change. 😢  They are randomly assigned by HEOS and seem to be have memory-based persistence only. When firmware updates, or all HEOS devices reset, new IDs are generated. HEOS provides no other reliable unique identifier, so we try to match by name and update.